src/averageAges.js

```diff
@@ -1,61 +1,70 @@
 'use strict';
 
 /**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
  * @param {object[]} people
  * @param {number} century - optional
  *
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const sortByCentury = century
+    ? people.filter((person) => Math.ceil(person.died / 100) === century)
+    : people;
+
+  const sortByGender = sortByCentury.filter((person) => person.sex === 'm');
+
+  const averageAge = sortByGender.reduce((acc, person) => (
+    acc + (person.died - person.born)), 0)
+    / sortByGender.length;
+
+  return averageAge;
 }
 
 /**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
  * @param {object[]} people
  * @param {boolean} withChildren - optional
  *
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter((person) => person.sex === 'f');
+  const motherNames = people.map((person) => person.mother);
+  const womenWithChildren = women.filter((person) => (
+    motherNames.includes(person.name))
+  );
+
+  const averageAge = withChildren
+    ? (womenWithChildren
+      .reduce((acc, person) => acc + (person.died - person.born), 0))
+      / womenWithChildren.length
+    : (women.reduce((acc, person) => acc + (person.died - person.born), 0)
+      / women.length
+    );
```

Comment:
Ternary is for simple data.
Do all logic inside `reduce` instead of making this monstrous ternary.